server/middleware/auth_middleware.go

```diff
@@ -1,54 +1,25 @@
 package middleware
 
 import (
-	"fmt"
-	"log"
-
-	"github.com/bokoness/lashon/models"
 	"github.com/bokoness/lashon/services"
 	"github.com/gofiber/fiber/v2"
 )
 
-func InsertUserToSession(c *fiber.Ctx, user *models.User) int {
-	sess, err := services.GetStore(c)
-	statusCodeError := fiber.StatusInternalServerError
-	if err != nil {
-		return statusCodeError
-	}
-
-	sess.Set("user", user)
-
-	if err = sess.Save(); err != nil {
-		log.Fatal(fmt.Printf("this is the error %s", err))
-		return statusCodeError
-	}
-
-	return 0
-}
-func GetUserFromCookiesByJWT(c *fiber.Ctx) (*models.User, error) {
-	cookie := c.Cookies("user")
-
-	if cookie == "" {
-		return nil, fiber.NewError(fiber.StatusUnauthorized)
-	}
 
-	user, err := services.GetUserByJWT(cookie)
-	return user, err
-}
 func AuthMiddleware(c *fiber.Ctx) error {
-	user, err := GetUserFromCookiesByJWT(c)
+	user, err := services.GetUserFromCookiesByJWT(c)
 	if err != nil || user == nil {
 		return c.SendStatus(fiber.StatusUnauthorized)
 	}
 
 	// insert user to session
-	statusCodeError := InsertUserToSession(c, user)
+	statusCodeError := services.InsertUserToSession(c, user)
```

Comment:
As I mentioned before, that service shouldn't know anything about status code, its the responsibility of the controller, so the service should return only error, no statusCode.